### DISCUSSION TITLE
Bouncing on edges

### DISCUSSION THREAD

**Discussion by Pindrought** [jrouwe/JoltPhysics#874] [Q&A] [answered]
I believe what i'm encountering is the same issue as
https://github.com/jrouwe/JoltPhysics/issues/717

I am wondering if this is expected behavior or not. 

https://github.com/jrouwe/JoltPhysics/assets/39708313/5a5d571d-645f-41a6-a2d0-9edd18b9f0fd

My scene is generated from a bunch of modular parts put together, and it seems like the dividers on the sides are causing some strange behaviors on the edges where it bounces the body up.

Code example is a replacement of the CharacterBaseTest.cpp. I had to use google drive, because of the large file size due to all of the vertex/triangle data in the cpp file.
https://drive.google.com/file/d/1jh1QIgeI6na7rOLyN0dQ0n27M0OMqPX0/view?usp=sharing

**Comment by jrouwe** [comment, marked as answer]
Yes this is almost certainly the same thing:

![image](https://github.com/jrouwe/JoltPhysics/assets/1621693/bf118f2d-b5d9-4f1d-b6e8-f889e3f47ad3)

If you turn on "Drawing Options" / "Draw Mesh Shape Triangle Outlines" you can see that the edges between parts become 'active' (= green) meaning the capsule can collide with them. In response the velocity (= yellow arrow) gets a slight upward angle when crossing one of these edges which makes the movement a bit weird.

CharacterVirtual is much better at hiding this than Character, so if this is about characters only you can switch. Alternatively you could collapse the parts into a single mesh so that it can properly detect the active edges.

I have done some investigation towards #717 but I have nothing yet that is significantly better than the current implementation.

**Reply by TIHan** [reply]
I've noticed something similar happen using cubes. The cube will get stuck on the edges and won't smoothly move across. I guess "contact welding" would help alleviate that? I'm sure it's not trivial to add. CharacterVirtual doesn't have issues moving across though.

**Comment by jrouwe** [comment]
@TIHan Cubes will have the exact same problem.

I did some work on this, please look at #717 for more information.
